src/rust/engine/testutil/src/data.rs

```diff
@@ -0,0 +1,120 @@
+use bazel_protos;
+use bytes;
+use digest::FixedOutput;
+use hashing;
+use protobuf::Message;
+use sha2::{self, Digest};
+
+pub struct TestData {
+  string: String,
+}
+
+impl TestData {
+  pub fn roland() -> TestData {
+    TestData::new("European Burmese")
+  }
+
+  pub fn catnip() -> TestData {
+    TestData::new("catnip")
+  }
+
+  pub fn new(s: &str) -> TestData {
+    TestData {
+      string: s.to_owned(),
+    }
+  }
+
+  pub fn bytes(&self) -> bytes::Bytes {
+    bytes::Bytes::from(self.string.as_str())
+  }
+
+  pub fn fingerprint(&self) -> hashing::Fingerprint {
+    hash(&self.bytes())
+  }
+
+  pub fn digest(&self) -> hashing::Digest {
+    hashing::Digest(self.fingerprint(), self.string.len())
+  }
+}
+
+pub struct TestDirectory {
+  directory: bazel_protos::remote_execution::Directory,
+}
+
+impl TestDirectory {
+  pub fn containing_roland() -> TestDirectory {
```

Comment:
I'm thinking doc strings that make the structure of each test directory would make understanding this way easier for someone new looking at the code